src/main/java/com/google/devtools/common/options/OptionsParser.java

```diff
@@ -929,4 +934,18 @@ public static boolean getUsesOnlyCoreTypes(Class<? extends OptionsBase> optionsC
     OptionsData data = OptionsParser.getOptionsDataInternal(optionsClass);
     return data.getUsesOnlyCoreTypes(optionsClass);
   }
+
+  /**
+   * A container for an arg and associated options that should be silently ignored when parsed but
+   * not recognized by the current command.
+   */
+  public static final class ArgAndFallbackData {
```

Comment:
I can't use records in this target as it needs to be compatible with Java 11.

Reply:
I wasn't aware of any requirement to support java 11 still. Where are you seeing that?

Reply:
It comes from here: https://github.com/bazelbuild/bazel/blob/1845f827ca62cc50433eb86076f466b3767d0ed4/src/main/java/com/google/devtools/common/options/BUILD#L43

Reply:
Thanks, I asked around to confirm that it's still necessary, but I'll proceed with reviewing.